src/aap_eda/core/validators.py

```diff
@@ -235,3 +235,23 @@ def check_credential_types(
         raise serializers.ValidationError(
             f"The type of credential can only be one of {names}"
         )
+
+
+def check_credential_registry_username_password(
+    eda_credential_id: int,
+) -> int:
+    credential = get_credential_if_exists(eda_credential_id)
+    name = credential.credential_type.name
+    if name != enums.DefaultCredentialType.REGISTRY.value:
+        raise serializers.ValidationError(
+            "The type of credential can only be one of "
+            f"['{enums.DefaultCredentialType.REGISTRY.value}']"
+        )
```

Comment:
We should be able to simply use `enums.DefaultCredentialType.REGISTRY`.

If not that's an issue but unrelated to these changes.